### PR TITLE
fix(protocol): Native u64

### PR DESCRIPTION
**Pull request by refcell**
### Description

Uses the native `u64` type - this prevents a lot of type conversions in kona.

See https://github.com/anton-rs/kona/pull/495

### CODE REVIEW

File: crates/protocol/src/block.rs [hunk 1]
```diff
@@ -53,12 +55,39 @@ pub struct L2BlockInfo {
     /// The L1 origin [BlockID]
     pub l1_origin: BlockID,
     /// The sequence number of the L2 block
-    pub seq_num: U64,
+    #[cfg_attr(feature = "serde", serde(deserialize_with = "deserialize_u64_hex"))]
+    pub seq_num: u64,
```
Comment (Rjected) [inline crates/protocol/src/block.rs:59]: I think all of these can be replaced with
```
    #[serde(with = "alloy_serde::quantity")]
```

**Review by Rjected — CHANGES_REQUESTED** [review]
one comment on alloy serde

File: crates/protocol/src/block.rs [hunk 2]
```diff
@@ -53,12 +55,39 @@ pub struct L2BlockInfo {
...
+}
+
+#[cfg(feature = "serde")]
+fn deserialize_u64_hex<'de, D>(deserializer: D) -> Result<u64, D::Error>
```
Comment (mattsse) [inline crates/protocol/src/block.rs:63]: yeah this is just alloy_serde::quantity

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/protocol/src/block.rs (hunk 1 above)
Comment (refcell) [reply crates/protocol/src/block.rs:59]: Amazing thank you

**Review by Rjected — APPROVED** [review]
nice

**Pull request merged**